src/gazebo_gps_plugin.cpp

```diff
@@ -39,13 +39,23 @@ GpsPlugin::~GpsPlugin()
 }
 
 bool GpsPlugin::checkWorldHomePosition(physics::WorldPtr world) {
-  if(!world_->SphericalCoords()) {
+  common::SphericalCoordinatesPtr spherical_coords;
+
+#if GAZEBO_MAJOR_VERSION >= 9
+  spherical_coords = world_->SphericalCoords();
+#else
+  spherical_coords = world_->GetSphericalCoordinates();
+#endif
+
+  if (!spherical_coords) {
     return false;
   }
-  double latitude = world_->SphericalCoords()->LatitudeReference().Degree();
-  double longitude = world_->SphericalCoords()->LongitudeReference().Degree();
-  double altitude = world_->SphericalCoords()->GetElevationReference();
-  if(latitude || longitude || altitude ) {
+  const double latitude = spherical_coords->LatitudeReference().Degree();
+  const double longitude = spherical_coords->LongitudeReference().Degree();
```

Comment:
You can just store the LLA as class member variables and use it bellow. Otherwise you are just repeating the same calls.